lib/synthesize-stream.ts

```diff
@@ -126,17 +126,42 @@ class SynthesizeStream extends Readable {
 
     socket.onmessage = message => {
       const chunk = message.data;
-      // some messages are strings - emit those unencoded, but push them to
-      // the stream as binary
-      const data = typeof chunk === 'string' ? chunk : Buffer.from(chunk);
+      // some info messages are sent as strings, telling the content_type and
+      // timings. Emit them as separate events, but do not send them along the
+      // pipe.
+      if (typeof chunk === 'string') {
+        try {
+          const json = JSON.parse(chunk);
+          if (json['binary_streams']) {
+            self.emit('binary_streams', message, chunk);
+          }
+          else if (json['marks']) {
+            self.emit('marks', message, chunk);
+          }
+          else if (json['words']) {
+            self.emit('words', message, chunk);
+          }
+          else if (json['error']) {
+            self.emit('error', message, chunk);
+          }
+          else if (json['warnings']) {
+            self.emit('warnings', message, chunk);
+          }
+        }
+        finally {
+          self.emit('message', message, chunk);
+        }
+        return;
+      }
+
       /**
        * Emit any messages received over the wire, mainly used for debugging.
        *
        * @event SynthesizeStream#message
        * @param {Object} message - frame object received from service
        * @param {Object} data - a data attribute of the frame that's either a string or a Buffer/TypedArray
        */
-      self.emit('message', message, data);
+      self.emit('message', message, chunk);
```

Comment:
Technically, this will be a different result because before, `data` would have been `Buffer.from(chunk)`. Is there a reason we wouldn't want to return this converted to a Buffer? As in, is this intended to be an improvement in behavior? If not, I would want to keep the behavior the same

Reply:
You are right, I mistakenly thought it would always send either a string or Buffer, but it sends a `ArrayBuffer` in actuality. I've fixed this so that these two lines will get a `Buffer.from(chunk)` like it was before.